phoenix-core/src/main/java/org/apache/phoenix/calcite/CalciteUtils.java

```diff
@@ -1036,4 +1048,40 @@ public Void visitCall(RexCall call) {
             return null;
         }
     }
+
+    public static Object convertLiteral(SqlLiteral literal, PhoenixRelImplementor implementor) {
+        try {
+            final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+            final FrameworkConfig config = Frameworks.newConfigBuilder()
+                    .parserConfig(SqlParser.Config.DEFAULT)
+                    .defaultSchema(rootSchema).build();
+            Planner planner = Frameworks.getPlanner(config);
+
+            SqlParserPos POS = SqlParserPos.ZERO;
+            final SqlNodeList selectList =
+                    new SqlNodeList(
+                            Collections.singletonList(literal),
+                            SqlParserPos.ZERO);
+
+
+            String sql = new SqlSelect(POS, SqlNodeList.EMPTY, selectList, null, null, null, null,
```

Comment:
I tried desperately to do this, but Calcite has pretty strict enforcement on Sql States. When the node is created its in STATE 1: READY, and it will not allow you to go straight to a validation state. Is there some way of overriding this?